### PR TITLE
log Admin changes to AuditTrail

### PR DESCRIPTION
**Pull request by annashamray**
issue - https://github.com/open-zaak/open-zaak/issues/169

### CODE REVIEW

File: src/openzaak/utils/admin.py [hunk 1]
```diff
@@ -118,3 +121,58 @@ class DynamicArrayMixin:
...
+
+    def save_model(self, request, obj, form, change):
+        model = obj.__class__
+        basename = model._meta.object_name.lower()
+
+        viewset = self.get_viewset(request)
+        version, scheme = viewset.determine_version(request, version=api_settings.DEFAULT_VERSION, uuid=obj.uuid)
+        request.version, request.versioning_scheme = version, scheme
+
+        action = 'update' if change else 'create'
```
Comment (joeribekker) [inline src/openzaak/utils/admin.py]: Isn't this an enum somewhere?

Comment (annashamray) [reply src/openzaak/utils/admin.py]: yeah, `CommonResourceAction`, I'll change it. It's a bit raw now.

File: src/openzaak/components/besluiten/tests/admin/test_besluit_audittrail.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        self.assertEqual(audittrail.hoofd_object, f"http://testserver{besluit_url}"),
+        self.assertEqual(audittrail.resource, "besluit"),
+        self.assertEqual(audittrail.resource_url, f"http://testserver{besluit_url}"),
+        self.assertEqual(audittrail.resource_weergave, besluit.unique_representation()),
+
+        old_data, new_data = audittrail.oud, audittrail.nieuw
+        self.assertEqual(old_data["toelichting"], "old")
+        self.assertEqual(new_data["toelichting"], "new")
+
+    def test_delete_besluit_action(self):
```
Comment (stevenbal) [inline src/openzaak/components/besluiten/tests/admin/test_besluit_audittrail.py]: This means that the AuditTrail remains in the database, even after the main resource (`Zaak` in this case) is deleted right? Where would you be able to see the audittrail for this action, if the resource is already deleted?

@joeribekker @sergei-maertens currently in the audittrail in vng-api-common the audittrail is deleted if the main resource is deleted via the API, is this okay?

Comment (joeribekker) [reply src/openzaak/components/besluiten/tests/admin/test_besluit_audittrail.py]: > This means that the AuditTrail remains in the database, even after the main resource (Zaak in this case) is deleted right? Where would you be able to see the audittrail for this action, if the resource is already deleted?

This is indeed not correct. The audit trail should be deleted if the main resource is deleted.

> @joeribekker @sergei-maertens currently in the audittrail in vng-api-common the audittrail is deleted if the main resource is deleted via the API, is this okay?

Yes. Delete really means destroy (even if I don't like this from a developer perspective). It's the law :)

**Comment by joeribekker** [comment]
@annashamray Can you make it so the deletion of the main resource will remove the audit trail as well? I'm not sure how this is done via the API but I thought it was just a cascade delete operation... (evidently, its not).

**Pull request merged**